File: app/src/main/java/fi/kaupunkifillarit/analytics/LocationPermissionsEvents.java
package fi.kaupunkifillarit.analytics;

import com.google.android.gms.analytics.HitBuilders;

import java.util.Map;

public abstract class LocationPermissionsEvents {
    private static final String CATEGORY = "Location Permissions";

    public static Map<String, String> granted() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_PERMISSION_GRANTED)
                .setValue(1)
                .build();
    }

    public static Map<String, String> denied() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_PERMISSION_DENIED)
                .setValue(1)
                .build();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/analytics/LocationPermissionsEvents.kt
package fi.kaupunkifillarit.analytics

import com.google.android.gms.analytics.HitBuilders

object LocationPermissionsEvents {
    private const val CATEGORY = "Location Permissions"

    fun granted(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_PERMISSION_GRANTED)
                .setValue(1)
                .build()
    }

    fun denied(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_PERMISSION_DENIED)
                .setValue(1)
                .build()
    }
}
